### PR TITLE
More PY3 fixes in test suite

### PR DESCRIPTION
**Pull request by terminalmage**
This fixes a couple netapi tests, and also adds the needed bits in runtests.py to make --grains tests actually run when that argument is passed.

@whiteinge can you have a look at the netapi test fixes? One involved a response coming back from cherrypy as a bytestring (which ``json.dumps()`` did not like), and another involved the event from tornado containing bytestrings. I assumed these were expected changes to the return in both cases, so I updated the tests to deal with the fact that the reponses come back containing bytestrings. I just want to make sure that is the correct fix.

### CODE REVIEW

File: tests/integration/netapi/rest_cherrypy/test_app.py [hunk 1]
```diff
@@ -190,7 +193,7 @@ def test_accepts_arg_kwarg_keys(self):
                 'Accept': 'application/json',
             }
         )
-        resp = json.loads(response.body[0])
+        resp = json.loads(salt.utils.to_str(response.body[0]))
```
Comment (s0undt3ch) [inline tests/integration/netapi/rest_cherrypy/test_app.py]: Get the encoding from the headers and use that. to_str will use the filesystem encoding which is not what we need here.

File: tests/integration/netapi/rest_tornado/test_app.py [hunk 2]
```diff
@@ -555,7 +556,10 @@ def verify_event(future):
             event = future.result()
             self.assertEqual(event['tag'], 'salt/netapi/hook')
             self.assertIn('headers', event['data'])
-            self.assertEqual(event['data']['post'], {'foo': 'bar'})
+            self.assertEqual(
+                event['data']['post'],
+                {'foo': salt.utils.to_bytes('bar')}
```
Comment (s0undt3ch) [inline tests/integration/netapi/rest_tornado/test_app.py]: This is probably safe here since its the event system, in our control.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/integration/netapi/rest_cherrypy/test_app.py (hunk 1 above)
Comment (terminalmage) [inline tests/integration/netapi/rest_cherrypy/test_app.py]: The encoding is not in the headers, here's a dump of the headers:

```python
{'Access-Control-Allow-Credentials': 'true',
 'Access-Control-Allow-Origin': '*',
 'Access-Control-Expose-Headers': 'GET, POST',
 'Allow': 'GET, HEAD, POST',
 'Cache-Control': 'private',
 'Content-Length': 65,
 'Content-Type': 'application/json',
 'Date': 'Thu, 23 Mar 2017 00:29:20 GMT',
 'Server': 'CherryPy/10.2.1',
 'Vary': 'Accept-Encoding'}
```

**Review by whiteinge — APPROVED** [review]
@s0undt3ch's suggestion is a good one but I'm not sure the faux test server will have that encoding header present. (My dev env isn't working ATM to verify.)

I'm still pretty unknowledgeable about the move to bytestrings in Py3 but I pulled down the latest develop (f1686009b3ff) to test those changes on Py2 and everything looks to be working as normal. So if this change fixes the test then it's ok by me. ¯\_(ツ)_/¯

File: tests/integration/netapi/rest_cherrypy/test_app.py (hunk 1 above)
Comment (s0undt3ch) [inline tests/integration/netapi/rest_cherrypy/test_app.py]: Wait! Wait!
Sorry @terminalmage, I was being dumb.
The content type is  'application/json', so it must be UTF-8, and in this case, pass it explicitly.

**Comment by s0undt3ch** [comment]
The failures are unrelated, and my suggestion to pass utf-8 explicitly is not a blocker. Merging.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**